File: android/Reguerta/app/src/main/java/com/reguerta/user/domain/notifications/NotificationEvent.kt
package com.reguerta.user.domain.notifications

import com.reguerta.user.domain.access.Member
import com.reguerta.user.domain.access.MemberRole

data class NotificationEvent(
    val id: String,
    val title: String,
    val body: String,
    val type: String,
    val target: String,
    val userIds: List<String>,
    val segmentType: String?,
    val targetRole: MemberRole?,
    val createdBy: String,
    val sentAtMillis: Long,
    val weekKey: String?,
) {
    fun isVisibleTo(member: Member): Boolean =
        when (target) {
            "all" -> true
            "users" -> userIds.contains(member.id)
            "segment" -> when (segmentType) {
                "role" -> targetRole?.let { member.roles.contains(it) } ?: false
                else -> false
            }
            else -> false
        }
}

File: ios/Reguerta/Reguerta/Domain/Notifications/NotificationEvent.swift
import Foundation

enum NotificationContentPolicy: Equatable {
    case embedded
    case authorizedFetchRequired
}

struct NotificationEvent: Identifiable, Equatable {
    let id: String
    let title: String
    let body: String
    let type: String
    let target: String
    let userIds: [String]
    let segmentType: String?
    let targetRole: MemberRole?
    let createdBy: String
    let sentAtMillis: Int64
    let weekKey: String?
    let contentPolicy: NotificationContentPolicy

    func isVisible(to member: Member) -> Bool {
        switch target {
        case "all":
            return true
        case "users":
            return userIds.contains(member.id)
        case "segment":
            guard segmentType == "role", let targetRole else { return false }
            return member.roles.contains(targetRole)
        default:
            return false
        }
    }
}

extension NotificationEvent {
    init(
        id: String,
        title: String,
        body: String,
        type: String,
        target: String,
        userIds: [String],
        segmentType: String?,
        targetRole: MemberRole?,
        createdBy: String,
        sentAtMillis: Int64,
        weekKey: String?
    ) {
        self.init(
            id: id,
            title: title,
            body: body,
            type: type,
            target: target,
            userIds: userIds,
            segmentType: segmentType,
            targetRole: targetRole,
            createdBy: createdBy,
            sentAtMillis: sentAtMillis,
            weekKey: weekKey,
            contentPolicy: .embedded
        )
    }
}
